tests/integration/src/setup.rs

```diff
@@ -224,44 +261,84 @@ pub const BLOCK_TIME: u64 = 12_000;
 
 pub fn run_to_block(n: BlockNumber) {
     while System::block_number() < n {
-        let block_number = System::block_number();
-        TransactionPayment::on_finalize(block_number);
-        Oracle::on_finalize(block_number);
-        PriceAggregator::on_finalize(block_number);
-        DappStaking::on_finalize(block_number);
-        Authorship::on_finalize(block_number);
-        Session::on_finalize(block_number);
-        AuraExt::on_finalize(block_number);
-        PolkadotXcm::on_finalize(block_number);
-        Ethereum::on_finalize(block_number);
-        CollatorSelection::on_finalize(block_number);
-        DynamicEvmBaseFee::on_finalize(block_number);
-        Inflation::on_finalize(block_number);
-
-        System::set_block_number(block_number + 1);
-        let block_number = System::block_number();
+        on_finalize();
 
-        Inflation::on_initialize(block_number);
-        Timestamp::set_timestamp(block_number as u64 * BLOCK_TIME);
-        TransactionPayment::on_initialize(block_number);
-        DappStaking::on_initialize(block_number);
-        Oracle::on_initialize(block_number);
-        PriceAggregator::on_initialize(block_number);
-        Authorship::on_initialize(block_number);
-        Aura::on_initialize(block_number);
-        AuraExt::on_initialize(block_number);
-        CollatorSelection::on_initialize(block_number);
-        Ethereum::on_initialize(block_number);
-        DynamicEvmBaseFee::on_initialize(block_number);
-        #[cfg(any(feature = "shibuya", feature = "shiden"))]
-        RandomnessCollectiveFlip::on_initialize(block_number);
+        System::set_block_number(System::block_number() + 1);
+        on_initialize();
```

Comment:
```suggestion
        AllPalletsWithSystem::on_initialize(block_number)
```

Reply:
same for other hooks

Reply:
Thanks, didn't know about those calls.

However, I'm not sure if the complexity of the code actually decreased with my changes 🥲